app/src/main/java/agency/five/codebase/android/movieapp/ui/moviedetails/MovieDetailsViewModel.kt

```diff
@@ -1,24 +1,41 @@
 package agency.five.codebase.android.movieapp.ui.moviedetails
 
 import agency.five.codebase.android.movieapp.data.repository.MovieRepository
-import agency.five.codebase.android.movieapp.mock.MoviesMock
 import agency.five.codebase.android.movieapp.ui.moviedetails.mapper.MovieDetailsMapper
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.launch
 
 class MovieDetailsViewModel(
     private val movieRepository: MovieRepository,
     movieDetailsMapper: MovieDetailsMapper,
     movieId: Int
 ) : ViewModel() {
-    val movieDetailsViewState: StateFlow<MovieDetailsViewState> = movieRepository.movieDetails(movieId)
-        .map { movies -> movieDetailsMapper.toMovieDetailsViewState(movies) }
-        .stateIn(viewModelScope, SharingStarted.Eagerly, movieDetailsMapper.toMovieDetailsViewState(MoviesMock.getMovieDetails(movieId)))
+    val movieDetailsViewState: StateFlow<MovieDetailsViewState> =
+        movieRepository.movieDetails(movieId)
+            .map { movies -> movieDetailsMapper.toMovieDetailsViewState(movies) }
+            .stateIn(
+                viewModelScope,
+                SharingStarted.WhileSubscribed(1000L),
+                MovieDetailsViewState(
+                    id = 0,
+                    imageUrl = null,
+                    voteAverage = 0.0F,
+                    title = "",
+                    overview = "",
+                    isFavorite = false,
+                    crew = listOf(),
+                    cast = listOf()
+                )
+            )
 
     fun toggleFavorite(movieId: Int) {
-        viewModelScope.launch {
+        viewModelScope.launch(Dispatchers.IO) {
```

Comment:
Why did you add `Dispatchers.IO` to all viewModelScope launches?

Reply:
Can't access db on main thread, getting IllegalStateException.

Reply:
This shouldn't be an issue, but it is okay to leave it like this